src/modules/licences/routes/agreements.js

```diff
@@ -29,5 +29,22 @@ module.exports = {
         }
       }
     }
+  },
+
+  deleteAgreement: {
+    method: 'DELETE',
+    path: '/water/1.0/agreements/{agreementId}',
+    handler: controller.deleteAgreement,
+    config: {
+      description: 'Deletes the agreement with the specified id',
+      validate: {
+        params: {
+          agreementId: Joi.string().uuid().required()
+        },
+        headers: async values => {
```

Comment:
Due to some recent tweaks I made to the internalCallingUser plugin, it's fairly easy now to restrict access for a route if needed to only allow users with a particular scope (similar to the UI project)
https://github.com/DEFRA/water-abstraction-service/blob/develop/src/modules/charge-versions/routes/charge-version-workflows.js#L22